client/editor/map_editor.cpp

```diff
@@ -107,33 +118,78 @@ void map_editor::check_open()
  */
 void map_editor::close()
 {
-  // Notify the server we are exiting edit mode
+  // Notify the server we are exiting edit mode and show all floating widgets
   struct connection *my_conn = &client.conn;
   dsend_packet_edit_mode(my_conn, false);
-
-  setVisible(false);
   queen()->mapview_wdg->show_all_fcwidgets();
   queen()->unitinfo_wdg->show();
+  setVisible(false);
 }
 
 /**
- * \brief Populate a combo box with the current list of players
+ * \brief Populate a combo box with the current list of players by nation
+ * name
  */
 void map_editor::update_players()
 {
   if (!players_done) {
+
+    // Ensure we can take all possible nations
+    send_chat_printf("/set allowtake hHaAoObd");
+
     players_iterate(pplayer)
     {
       auto sprite =
           *get_nation_flag_sprite(tileset, nation_of_player(pplayer));
-      ui.combo_players->addItem(sprite,
-                                _(qUtf8Printable(player_name(pplayer))));
+      ui.combo_players->addItem(
+          sprite, _(qUtf8Printable(nation_adjective_for_player(pplayer))));
     }
     players_iterate_end;
+
+    // Pirates and Barbarians are problematic, so we get rid of them
+    for (int i = 0; i <= ui.combo_players->count(); i++) {
+      if (ui.combo_players->itemText(i).toUtf8() == "Pirate") {
+        ui.combo_players->removeItem(i);
+      }
+      if (ui.combo_players->itemText(i).toUtf8() == "Barbarian") {
+        ui.combo_players->removeItem(i);
+      }
+    }
```

Comment:
This forgets the `Animal Kingdom` and will be problematic with translations or rulesets that redefine barbarian nations. Use `if (!is_barbarian(pplayer))` in the loop before.

Reply:
Oh nice. I wasn't aware of the function.